### DISCUSSION TITLE
Help fixing syntax errors in C++

### DISCUSSION THREAD

**Discussion by ghost** [community/community#122607] [Programming Help]
### Body

```cpp
#include <iostream>
#include <map>
#include <string>
using namespace std;
int main(){
    int index = 0;
    int line = 1; 
    string value;
    string array[1000] = {};
    string cmds[11] = {"+", "-", "(", ")", "?", ">", "&", "i", "s", "c", "f"};
    std::map<string, string>commands = {
        {"+", "Increment the index by 1"},
        {"-", "Decrement the index by 1"},
        {"(", "Move the current value in the current array's index to the stack"},
        {")", "Remove the current value in the current array's index to the stack"},
        {"?", "Change the value string's value"},
        {">", "Write the value string to the current array's index"},
        {"&", "Remove the current value in the current array's index from the stack"},
        {"i", "Store an integer into a specified register"},
        {"s", "Store a string into a specified register"},
        {"c", "Store a character into a specified register"},
        {"f", "Store a float into a specified register"},
    };
    while(true){
        cout << line << " > ";
        char inp = getchar();
        line = line + 1;
        cin.ignore();
        switch(inp){
            case '+':
                if(index < 1001){
                    index = index + 1;
                }else{
                    cout << "The index is at its max value!" << endl;
                }
                break;
            case '-':
                if(index > 0){
                    index = index - 1;
                }else{
                    cout << "The index is already empty!" << endl;
                }
                break;
            case ')':
                string val = array[index];
                asm{
                    PUSH val
                
                break;
            case '(':
                asm{
                    POP val
                }
                break;
            case 'i':
                int val;
                string Register;
                cout << "Enter an integer value > ";
                cin >> val;
                cout << "Enter a register > ";
                cin >> Register;
                if(Register == "EAX"){
                    __asm{
                        MOV EAX, val
                    }
                }else if(Register == "EBX"){
                    __asm{
                        MOV ECX, val
                    }
                }else if(Register == "ECX"){
                    __asm{
                        MOV ECX, val
                    }
                }else if(Register == "EDX"){
                    __asm{
                        MOV EDX, val
                    }
                }else if(Register == "ESI"){
                    __asm{
                        MOV ESI, val 
                    }
                }else{
                    cout << Register << " is not a valid register!" << endl;
                }
            case 'c':
                char val;
                string Register;
                cout << "Enter a character value > ";
                cin >> val;
                cout << "Enter a register > ";
                cin >> Register;
                if(Register == "EAX"){
                    __asm{
                        MOV EAX, val 
                    }
                }else if(Register == "EBX"){
                    __asm{
                        MOV EBX, val
                    }
                }else if(Register == "ECX"){
                    __asm{
                        MOV ECX, val
                    }
                }else if(Register == "EDX"){
                    __asm{
                        MOV EDX, val
                    }
                }else if(Register == "ESI"){
                    __asm{
                        MOV ESI, val
                    }
                }else{
                    cout << Register << " is not a valid register!" << endl;
                }
            case 's':
                string val;
                string Register;
                cout << "Enter a string value > ";
                cin >> val;
                cout << "Enter a register > ";
                cin >> Register;
                if(Register == "EAX"){
                    __asm{
                        MOV EAX, val
                    }
                }else if(Register == "EBX"){
                    __asm{
                        MOV EBX, val
                    }
                }else if(Register == "ECX"){
                    __asm{
                        MOV ECX, val
                    }
                }else if(Register == "EBX"){
                    __asm{
                        MOV EDX, val
                    }
                }else if(Register == "EDX"){
                    __asm{
                        MOV EDX, val
                    }
                }else if(Register == "ESI"){
                    __asm{
                        MOV ESI, val
                    }
                }else{
                    cout << Register << " is not a valid register" << endl;
                }
            default:
                if(inp != ""){
                    cout << "Syntax error at line " << line << " (" << inp << ")" << "." << endl;
                }else{
                    continue;
                }
        }
    }
    return 0;
}```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Vivek09Chahal** [comment]
Good Evening, I have looked at your code and it is a bit meshy. And I wrote it again, but I don't know exact assembly language so here is code have some fixes you can check.

`
#include <iostream>
#include <map>
#include <string>

using namespace std;

int main()
{
    int index = 0;
    int line = 1;
    string value;
    string array[1000] = {};
    string cmds[11] = {"+", "-", "(", ")", "?", ">", "&", "i", "s", "c", "f"};
    std::map<string, string> commands = {
        {"+", "Increment the index by 1"},
        {"-", "Decrement the index by 1"},
        {"(", "Move the current value in the current array's index to the stack"},
        {")", "Remove the current value in the current array's index to the stack"},
        {"?", "Change the value string's value"},
        {">", "Write the value string to the current array's index"},
        {"&", "Remove the current value in the current array's index from the stack"},
        {"i", "Store an integer into a specified register"},
        {"s", "Store a string into a specified register"},
        {"c", "Store a character into a specified register"},
        {"f", "Store a float into a specified register"},
    };
    while (true)
    {
        cout << line << " > ";
        char inp = getchar();
        line = line + 1;
        cin.ignore();

        int val;
        string Register;
        cout << "Enter a character value > ";
        cin >> val;
        cout << "Enter a register > ";
        cin >> Register;

        switch (inp)
        {
        case '+':
        {
            if (index < 1001)
            {
                index = index + 1;
            }
            else
            {
                cout << "The index is at its max value!" << endl;
            }
            break;
        }
        case '-':
        {
            if (index > 0)
            {
                index = index - 1;
            }
            else
            {
                cout << "The index is already empty!" << endl;
            }
            break;
        }
        case ')':
        {
            string val = array[index];
            __asm (
                    PUSH val;
            );
            break;
        }
        case '(':
        {
            __asm (
                    POP val;
            );
            break;
        }
        case 'i':
        {
            if (Register == "EAX")
            {
                __asm (
                        MOV EAX, val;
                );
            }
            else if (Register == "EBX")
            {
                __asm (
                        MOV ECX, val;
                );
            }
            else if (Register == "ECX")
            {
                __asm (
                        MOV ECX, val;
                );
            }
            else if (Register == "EDX")
            {
                __asm (
                        MOV EDX, val;
                );
            }
            else if (Register == "ESI")
            {
                __asm (
                        MOV ESI, val;
                );
            }
            else
            {
                cout << Register << " is not a valid register!" << endl;
            }
            break;
        }
        case 'c':
        {
            if (Register == "EAX")
            {
                __asm (
                    MOV EAX, val
                );
            }
            else if (Register == "EBX")
            {
                __asm (
                        MOV EBX, val;
                );
            }
            else if (Register == "ECX")
            {
                __asm (
                        MOV ECX, val;
                );
            }
            else if (Register == "EDX")
            {
                __asm (
                        MOV EDX, val;
                );
            }
            else if (Register == "ESI")
            {
                __asm (
                        MOV ESI, val;
                );
            }
            else
            {
                cout << Register << " is not a valid register!" << endl;
            }
            break;
        }
        case 's':
        {
            if (Register == "EAX")
            {
                __asm (
                        MOV EAX, val;
                );
            }
            else if (Register == "EBX")
            {
                __asm (
                        MOV EBX, val;
                );
            }
            else if (Register == "ECX")
            {
                __asm (
                        MOV ECX, val;
                );
            }
            else if (Register == "EBX")
            {
                __asm (
                        MOV EDX, val;
                );
            }
            else if (Register == "EDX")
            {
                __asm (
                        MOV EDX, val;
                );
            }
            else if (Register == "ESI")
            {
                __asm (
                        MOV ESI, val;
                );
            }
            else
            {
                cout << Register << " is not a valid register" << endl;
            }
            break;
        }
        default:
        {
            if (inp != '\0')
            {
                cout << "Syntax error at line " << line << " (" << inp << ")"
                     << "." << endl;
            }
            else
            {
                continue;
            }
        }
        }
        return 0;
    }
}
`

**Reply by ghost** [reply]
Thank you!
